lib/rate-limiter.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const _get = require('lodash.get');
+
+exports.register = (defaultRate, routes) => {
+  const rateTable = {};
+
+  routes.forEach((route) => {
+    if (!(route.method === 'get' || route.method === 'post' || route.method === 'delete')) {
+      return;
+    }
+
+    if (_get(route, 'settings.plugins.rateLimit.enabled') === false) {
```

Comment:
Right but you're still making that check down below with `if (plugins && plugins.rateLimit) {`. All I'm saying is bringing that check up a bit and utilizing the fact that you made sure certain properties exist previously, so you don't have to check again. Does that make sense?